plugins/history_plugin/history_plugin.cpp

```diff
@@ -112,7 +112,9 @@ namespace eosio {
 
    static void add(chainbase::database& db, const vector<permission_level_weight>& controlling_accounts, const account_name& account_name, const permission_name& permission)
    {
-      for (auto controlling_account : controlling_accounts ) {
+      std::set<permission_level_weight> controlling_accounts_set(controlling_accounts.begin(), controlling_accounts.end());
+
+      for (auto controlling_account : controlling_accounts_set ) {
```

Comment:
Sorry for the bit of misdirection here. The idea is to avoid uniqueness constraint on `account_control_history_object`. I think the most direct and easiest approach is to first find by   `[controlled_account, controlled_permission, controlling_account]` and only `create` if not found.